src/main/kotlin/com/cjbooms/fabrikt/generators/controller/SpringControllerInterfaceGenerator.kt

```diff
@@ -78,7 +79,8 @@ class SpringControllerInterfaceGenerator(
             .addSpringFunAnnotation(op, verb, path.pathString)
             .addSuspendModifier()
 
-        val funcSpec = if (options.contains(ControllerCodeGenOptionType.COMPLETION_STAGE)) {
+        val asyncSupport = op.extensions.containsKey(EXTENSION_ASYNC_SUPPORT)
+        val funcSpec = if (options.contains(ControllerCodeGenOptionType.COMPLETION_STAGE) || asyncSupport) {
```

Comment:
Should we allow asyncSupport to override default?
`x-async-support: false`

Reply:
Added the support for handling both true and false.